### ISSUE TITLE
Links to Cloud Native Glossary to become more newbie-friendly

### ISSUE THREAD

**Issue by CathPag** [kubernetes/website#39792]
**This is a Feature Request**

**What would you like to be added**

1. Link cloud native terms in the Kubernetes docs (including the Glossary) to [Cloud Native Glossary](https://glossary.cncf.io/) definitions.  
2. Update the docs contributor guide reminding contributors to link to the CN Glossary whenever feasible/beneficial.  

**Why is this needed**
Reading the docs can be overwhelming for users new to Kubernetes, as so many terms and concepts are new. While the Kubernetes docs do a great job at explaining lots of these terms, they are still quite advanced. The [Cloud Native Glossary](https://glossary.cncf.io/) assumes little to no context and could help ease new users' early Kubernetes learning journey. Hyperlinks to the CN Glossary would provide more context to newbies without disrupting the reading experience of more advanced users. 

Once implemented, it would be ideal if the CN Glossary would be on the radar of new doc contributors, so future iterations also link to it.

**Comments**
If the K8s maintainers agreed that this would be beneficial, I would also create a CN Glossary issue. This could be a great project for a new contributor. 

Thanks for considering!

**Comment by XDRAGON2002** [comment]
Hi @CathPag @sftim I'm a new contributor to the project, can I take this up? It seems like I'd just have to add links to the glossary for the definitions that exist in it?

**Comment by sftim** [comment]
Anyone is welcome to work on issues - see https://kubernetes.io/docs/contribute/ and https://www.kubernetes.dev/docs/ for tips.

Remember cover:

> 2. Update the docs contributor guide reminding contributors to link to the CN Glossary whenever feasible/beneficial.

in your change, as well as adding some links. In fact, that should be the first commit in your PR.

@XDRAGON2002 if you aren't sure what those terms mean - some are jargon - it's OK to start with a different change and revisit this when you're more confident.

/help

**Comment by krol3** [comment]
I would like to help too!

**Comment by MaryamTavakkoli** [comment]
@KhushPatibandha @krol3 Are you working on this?

**Comment by CathPag** [comment]
I believe anyone can just jump on this task and submit PRs for missing links, @MaryamTavakkoli. At least, I think that's what @sftim said. There are probably many different places where we can link to, more than one person would catch in one PR.

**Comment by krol3** [comment]
yeah! I'm agree @CathPag I will try to create a PR to have a sample here for the next contributors @MaryamTavakkoli

**Comment by vaibhav2107** [comment]
Is there someone who is working on this issue?

**Comment by kumarankit999** [comment]
I don't think someone is working on this issue still!!!
Go further @vaibhav2107 . 
Let me know if you need any help !!

**Comment by drewhagen** [comment]
We're going to take on some progress for this issue in the SIG Docs sprint at Contributor Summit NA 2024

**Comment by drewhagen** [comment]
### Linking Glossary Terms in Kubernetes Documentation

As a Kubernetes docs reader, you may encounter technical terms that feel confusing or unfamiliar, especially as a beginner. While glossaries exist, it would be more user-friendly to link these terms directly within the docs for easy reference.

There are two glossaries available for reference:
1. **CNCF Glossary**: Contains general, foundational terminology, suited for non-technical or general audiences.
2. **Kubernetes Glossary**: Offers more technical, specific definitions relevant to Kubernetes.

### Guidelines for Glossary Links

To improve readability and accessibility, we’ve established the following guidelines for linking terms within the documentation:

**Case 1: Term Exists in Both CNCF and Kubernetes Glossaries**
- **Primary Link**: Link to the Kubernetes glossary, as it typically provides a more specific definition.
- **Further Reading**: Within the Kubernetes glossary entry, include a link to the CNCF glossary if additional context is helpful.

**Case 2: Term Exists Only in Kubernetes Glossary**
- **Link**: Link directly to the Kubernetes glossary.

**Case 3: Term Exists Only in CNCF Glossary**
- **Link**: Link directly to the CNCF glossary.

These guidelines ensure that readers have easy access to the most relevant and comprehensive definitions, supporting a smoother learning experience in Kubernetes documentation.

**Comment by drewhagen** [comment]
Another idea we discussed at the Docs Sprint:

**IDEA**: Instead of Hyperlinks, could we use Wikipedia Style Footer?

Pod<sup>[1]</sup>
Node<sup>[2]</sup>
Service<sup>[3]</sup>

References:
1. For more general information, see: https://glossary.cncf.io/pod/
2. For more general information, see: https://glossary.cncf.io/node/
3. https://glossary.cncf.io/service/

**Comment by sftim** [comment]
> Another idea we discussed at the Docs Sprint:
> 
> **IDEA**: Instead of Hyperlinks, could we use Wikipedia Style Footer?

I wonder if we can make a shortcode for that. My guess is: yes we can.

(we might not need to, but it's an option)

Using a shortcode also lets us change the page rendering using `.HasShortcode`, which might come in handy.
